### PR TITLE
refactor(core) Make TRAM an installable package

### LINKED ISSUE #16: TRAM is a library
As a user, I want to use TRAM as a library, so I can get the key ML features without running the webapp.

Acceptance Criteria:

1. pip install tram works
2. ML pipeline is available via pip installed library

### PR DESCRIPTION
**Pull request by mehaase**
This looks like a big commit, but its actually one main change and a bunch of small changes to support that main change.

* add setup.cfg so that TRAM can be installed as a package. Now TRAM doesn't need to be manually added to PYTHONPATH, and also `python src/tram/manage.py` is now aliased to `tram` so you can do things like `tram runserver`.
* Update documentation and build artifacts to use `tram ...` shortcut instead of `python src/tram/manage.py ...`
* Move src/tram/tram/* to src/tram/*. This large rename causes the bulk of what you see in this commit, but I think its a cleaner layout, it's what developers will expect when they check out the project for the first time, and its a more natural fit for an installable package to be laid out this way.

Setting this as WIP until we can discuss in the tech lead group.

closes #16

### CODE REVIEW

**Comment by mehaase** [comment]
This refactor causes Sonar Cloud to find [a configuration vulnerability](https://sonarcloud.io/project/security_hotspots?id=center-for-threat-informed-defense_tram&pullRequest=155&resolved=false&types=SECURITY_HOTSPOT) in settings.py that I guess it couldn't detect in the old package layout?

File: Dockerfile [hunk 1]
```diff
@@ -45,10 +44,8 @@ RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/
 #ENV HTTP_PROXY=${http_proxy} \
 #    HTTPS_PROXY=${https_proxy} \
 #    NO_PROXY=${no_proxy}
-#COPY MY_ORG_ROOT.crt /usr/local/share/ca-certificates
-#RUN cd /usr/local/share/ca-certificates && \
-#    wget http://pki.my-organization.local/MY%20ORG%20ROOT.crt && \
-#    update-ca-certificates
+COPY ZScaler_Root.crt /usr/local/share/ca-certificates
+RUN update-ca-certificates
```
Comment (emmanvg) [inline Dockerfile:48]: Revert this back to the original commented commands... (I guess this is an oversight from local testing?)

**Comment by m3mike** [comment]
> @m3mike The super linter is throwing errors for the Django migrations, but pyproject.toml is supposed to ignore the migrations. Have you seen this before?

Yes I have, and yes it's annoying. If I remember right, since from a git perspective the file was changed, the file gets included in a changeset passed directly to the linters. And when a file list is presented to the linters, it can override the ignore regex. I'll add a suggestion thing to the ci check that should fix it.

**Comment by mehaase** [comment]
> Yes I have, and yes it's annoying. If I remember right, since from a git perspective the file was changed, the file gets included in a changeset passed directly to the linters. And when a file list is presented to the linters, it can override the ignore regex. I'll add a suggestion thing to the ci check that should fix it.

I was wondering if that was the issue, and I noticed this config `VALIDATE_ALL_CODEBASE: false`. What if I set that to true, then fix all of the issues it raises? (Can't be that many lint issues at this point since you linted the whole repo last week.)

**Comment by m3mike** [comment]
> > Yes I have, and yes it's annoying. If I remember right, since from a git perspective the file was changed, the file gets included in a changeset passed directly to the linters. And when a file list is presented to the linters, it can override the ignore regex. I'll add a suggestion thing to the ci check that should fix it.
> 
> I was wondering if that was the issue, and I noticed this config `VALIDATE_ALL_CODEBASE: false`. What if I set that to true, then fix all of the issues it raises? (Can't be that many lint issues at this point since you linted the whole repo last week.)



> 

Actually never tried that for this case but it could work, typically just ended up overriding the merge block on github and documented why in these cases. If you did go that route, you'd probably want to switch it back for normal use, though there isn't that much code so it might not increase CI time too much.

**Comment by mehaase** [comment]
`FILTER_REGEX_EXCLUDE` is exactly what I was looking for and the build is green again. Thanks @m3mike!

**Review by emmanvg — APPROVED** [review]
TRAM as a installable library works as expected. Linter issues resolved.

**Pull request merged**